### PR TITLE
shuffle pending accounts in the transaction pool

### PR DESCRIPTION
**Pull request by JekaMas**
Prevents possible vulnerability with pre-mined accounts

### CODE REVIEW

**Review by ssandeep — APPROVED** [review]
Should we also propose this change in geth?

**Review by temaniarpit27 — APPROVED** [review]

**Comment by temaniarpit27** [comment]
@JekaMas can you please look into the CI failure?

File: common/random/shuffle.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	//nolint:gosec
+	r := rand.New(CryptoSource{}) //CryptoSource is a cryptographic source of randomness
+
+	r.Shuffle(len(slice), func(i, j int) {
+		slice[i], slice[j] = slice[j], slice[i]
+	})
+}
+
+// ShuffleMap returns a shuffled slice of keys to iterate by
+func ShuffleMap[K comparable, V any](m map[K]V) []K {
```
Comment (cffls) [inline common/random/shuffle.go:17]: This isn't deterministically random I guess?

Comment (JekaMas) [reply common/random/shuffle.go:17]: As far as it uses crypto/rand, it isn't.

Comment (cffls) [reply common/random/shuffle.go:17]: This looks like a good explanation: https://stackoverflow.com/a/55925880

**Review by cffls — APPROVED** [review]

File: common/random/shuffle.go (hunk 1 above)
Comment (JekaMas) [reply common/random/shuffle.go:17]: Yeap, although it only depends on non cryptographically strong hash and a seed. The PR introduces cryptographically strong solution.

**Comment by 0xsharma** [comment]
@JekaMas Can you please have a look at the CI ? Thank you !

**Comment by temaniarpit27** [comment]
@JekaMas CI is failing. Can you please look into this?

**Comment by JekaMas** [comment]
I believe, we need to rethink whole idea of random ordering of transactions - https://github.com/ethereum/go-ethereum/issues/21350

**Comment by JekaMas** [comment]
Will redo with proving tests.

**Pull request closed without merging**